File: 프로그래머스/0/181870. ad 제거하기/ad 제거하기.py
def solution(strArr):
    answer = []
    contains = 'ad'
    
    for str in strArr:
        # if not str.find("ad") + 1: <- 비교가 이상해서 그냥 비교연산자로 수정
        # if str.find("ad") == -1: <- 가독성 안좋아서 수정
        if not contains in str:
            answer.append(str)
    
    return answer

File: 프로그래머스/0/181870. ad 제거하기/ad 제거하기.c
#include <stdio.h>
#include <stdbool.h>
#include <stdlib.h>

char** solution(const char* strArr[], size_t strArr_len) {
    char** answer = (char**)malloc(strArr_len * sizeof(char*));
    int ad_check = 0;
    int idx = 0;
    
    for(int i = 0; i < strArr_len; i++){
        ad_check = 0;
        for(int j = 0; j < strlen(strArr[i]) - 1; j++){
            if(strArr[i][j] == 'a' && strArr[i][j + 1] == 'd'){
                ad_check++;
                break;
            }
        }
        if(!ad_check){
            answer[idx++] = strArr[i];
        }
    }
    
    return answer;
}